mlxbf-bootctl.c

```diff
@@ -747,17 +750,76 @@ void correct_bootstream_headers(void *buf, int num_images) {
 // partition, based on a version number.
 // We do this conservatively, so we don't accidentally
 // filter out something common.
-bool should_install_image(boot_image_header_t *hdr, int version) {
+bool should_install_image(boot_image_header_t *hdr, int version, uint8_t *max_versions) {
   // version < 0 indicates filtering should be off. All
   // images will be installed in this case.
   if (version < 0)
     return true;
 
-  // For now, do not install anything that was introduced
-  // after the given version.
-  return hdr->cur_img_ver <= version;
+  // If version filtering is on, check max_versions is valid
+  if (max_versions == NULL)
+    die("internal: max_versions is NULL");
+
+  bool should_not_install = (
+    // Do not install anything introduced after given version
+    hdr->cur_img_ver > version ||
+
+    // Otherwise, check for redundancy.
+    // If the img version isn't the max, it shouldn't be installed.
+    hdr->cur_img_ver < max_versions[hdr->image_id]
```

Comment:
should it be "(hdr->cur_img_ver < max_versions[hdr->image_id]) && (max_versions[hdr->image_id] != 0)", which means
only to filtering old redundant images?